kbase-extension/static/kbase/js/narrativeLogin.js

```diff
@@ -26,25 +25,37 @@ define ([
     BootstrapDialog
 ) {
     'use strict';
-    var baseUrl = JupyterUtils.get_body_data('baseUrl'),
-        authClient = Auth.make({url: Config.url('auth')}),
-        sessionInfo = null,
-        tokenCheckTimer = null,
-        tokenWarningTimer = null;
+    const baseUrl = JupyterUtils.get_body_data('baseUrl');
+    const authClient = Auth.make({url: Config.url('auth')});
+    let sessionInfo = null;
+    let tokenCheckTimer = null;
+
+    // don't warn if less than 30 seconds to auto-logout
+    const ABOUT_TO_LOGOUT_MINIMUM = 1000 * 30;
+
+    // Do warn if les than 5 minutes to auto-logout
+    const ABOUT_TO_LOGOUT_THRESHOLD = 1000 * 60 * 5;
+
+    // How often to inspect the current browser auth token.
+    const TOKEN_MONITORING_INTERVAL = 1000;
+
+    // Delay after which to automatically logout the Narrative
+    // after the logging-out dialog is shown.
+    const AUTO_LOGOUT_DELAY = 1000 * 30;
 
     /* set the auth token by calling the kernel execute method on a function in
      * the magics module
      */
     function ipythonLogin(token) {
-        window.kb = new KBCacheClient(token);
+        window.kb = new window.KBCacheClient(token); // just as bad as global, but passes linting
         $.ajax({
```

Comment:
Yeah, I just had to do this to satisfy the linter.
I took a quick look at what it would take to have this loaded as an AMD module, but it was too hairy and thought satisfying the linter was good enough for today.